File: lib/models/lipid_profile.dart
class LipidProfile {
  final int id;
  final int userId;
  final DateTime testDate;
  final double totalCholesterol;
  final double hdl;
  final double ldl;
  final double vldl;
  final double triglycerides;
  final String? imageUrl;

  LipidProfile({
    required this.id,
    required this.userId,
    required this.testDate,
    required this.totalCholesterol,
    required this.hdl,
    required this.ldl,
    required this.vldl,
    required this.triglycerides,
    this.imageUrl,
  });

  // Factory constructor to create a LipidProfile instance from a map (JSON)
  factory LipidProfile.fromJson(Map<String, dynamic> json) {
    return LipidProfile(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      testDate: DateTime.parse(json['test_date'] as String),
      totalCholesterol: json['total_cholesterol'] as double,
      hdl: json['hdl'] as double,
      ldl: json['ldl'] as double,
      vldl: json['vldl'] as double,
      triglycerides: json['triglycerides'] as double,
      imageUrl: json['image_url'] as String?,
    );
  }

  // Method to convert a LipidProfile instance to a map (JSON)
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'test_date': testDate.toIso8601String(),
      'total_cholesterol': totalCholesterol,
      'hdl': hdl,
      'ldl': ldl,
      'vldl': vldl,
      'triglycerides': triglycerides,
      'image_url': imageUrl,
    };
  }
}

File: models-kotlin/LipidProfile.kt
package com.healthtracker.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class LipidProfile(
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("test_date")
    val testDate: LocalDateTime,
    @SerializedName("total_cholesterol")
    val totalCholesterol: Double,
    val hdl: Double,
    val ldl: Double,
    val vldl: Double,
    val triglycerides: Double,
    @SerializedName("image_url")
    val imageUrl: String? = null
) {
    companion object {
        private val dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME

        fun fromJson(json: Map<String, Any?>): LipidProfile {
            return LipidProfile(
                id = json["id"] as Int,
                userId = json["user_id"] as Int,
                testDate = LocalDateTime.parse(json["test_date"] as String, dateFormatter),
                totalCholesterol = json["total_cholesterol"] as Double,
                hdl = json["hdl"] as Double,
                ldl = json["ldl"] as Double,
                vldl = json["vldl"] as Double,
                triglycerides = json["triglycerides"] as Double,
                imageUrl = json["image_url"] as String?
            )
        }
    }

    fun toJson(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "user_id" to userId,
            "test_date" to testDate.format(dateFormatter),
            "total_cholesterol" to totalCholesterol,
            "hdl" to hdl,
            "ldl" to ldl,
            "vldl" to vldl,
            "triglycerides" to triglycerides,
            "image_url" to imageUrl
        )
    }
}